src/lib/mina_graphql/itn_zkapps.ml

```diff
@@ -0,0 +1,293 @@
+        (Uuid.to_string uuid) ;
+      Uuid.Table.remove scheduler_tbl uuid ;
+      true )
+    else if Ivar.is_full stop_signal then (
+      [%log info]
+        "Scheduled zkapp commands with handle %s received stop signal, stop \
+         deployment of zkapp accounts"
+        (Uuid.to_string uuid) ;
+      Uuid.Table.remove scheduler_tbl uuid ;
+      true )
+    else false
+  in
+  Deferred.List.iteri keypairs ~f:(fun i kp ->
+      let ndx = i mod num_fee_payers in
+      if finished () then Deferred.unit
+      else
+        let fee_payer_keypair = fee_payer_array.(ndx) in
+        let memo = sprintf "%s-%d" memo_prefix i in
+        let spec =
+          { Transaction_snark.For_tests.Deploy_snapp_spec.sender =
+              (fee_payer_keypair, !(fee_payer_nonces.(ndx)))
+          ; fee = deployment_fee
+          ; fee_payer = None
+          ; amount = init_balance
+          ; zkapp_account_keypairs = [ kp ]
+          ; memo = Signed_command_memo.create_from_string_exn memo
+          ; new_zkapp_account = true
+          ; snapp_update = Account_update.Update.dummy
+          ; preconditions = None
+          ; authorization_kind = Account_update.Authorization_kind.Signature
+          }
+        in
+        let zkapp_command =
+          Transaction_snark.For_tests.deploy_snapp ~constraint_constants
+            ~permissions:
+              ( if max_cost then
+                { Permissions.user_default with
+                  set_verification_key = Permissions.Auth_required.Proof
+                ; edit_state = Permissions.Auth_required.Proof
+                ; edit_action_state = Proof
+                }
+              else Permissions.user_default )
+            spec
+        in
+        let%bind () = after wait_span in
+        Deferred.repeat_until_finished ()
+        @@ fun () ->
+        if finished () then Deferred.return (`Finished ())
+        else
+          match%bind Zkapps.send_zkapp_command mina zkapp_command with
+          | Ok _ ->
+              fee_payer_nonces.(ndx) :=
+                Account.Nonce.succ !(fee_payer_nonces.(ndx)) ;
+              [%log info]
+                "Successfully submitted zkApp command that creates a zkApp \
+                 account"
+                ~metadata:
+                  [ ("zkapp_command", Zkapp_command.to_yojson zkapp_command) ] ;
+              Deferred.return (`Finished ())
+          | Error err ->
+              [%log info] "Failed to setup a zkApp account, try again"
+                ~metadata:
+                  [ ("zkapp_command", Zkapp_command.to_yojson zkapp_command)
+                  ; ("error", `String err)
+                  ] ;
+              let%bind () = after wait_span in
+              Deferred.return (`Repeat ()) )
+
+let is_zkapp_deployed kp ledger =
+  match
+    Option.try_with (fun () ->
+        let account = Utils.account_of_kp kp ledger in
+        Option.is_some account.zkapp )
+  with
+  | Some true ->
+      true
+  | _ ->
+      false
```

Comment:
```suggestion
Option.try_with (fun () ->
    let account = Utils.account_of_kp kp ledger in
    Option.is_some account.zkapp )
|> Option.value ~default:false
```

Reply:
I rewrote it to:

```ocaml
try Option.is_some (Utils.account_of_kp kp ledger).zkapp with _ -> false
```

Reply:
See commit [a7b61d2](https://github.com/MinaProtocol/mina/pull/14037/commits/a7b61d2c9e312b3994a65e4f1e7efd2869d282a6)